docs/blog/2021-01-25-zmk-sotf-4.md

```diff
@@ -0,0 +1,190 @@
+---
+title: "ZMK State Of The Firmware #4"
+author: Pete Johanson
+author_title: Project Creator
+author_url: https://gitlab.com/petejohanson
+author_image_url: https://www.gravatar.com/avatar/2001ceff7e9dc753cf96fcb2e6f41110
+tags: [SOTF, keyboards, firmware, oss, ble]
+---
+
+Welcome to the fourth ZMK "State Of The Firmware" (SOTF)!
+
+This update will cover all the major activity since [SOTF #3](/blog/2020/11/09/zmk-sotf-3).
+
+## Recent Activity
+
+Here's a summary of the various major changes since last time, broken down by theme:
+
+### Keymaps/Behaviors
+
+Since last time, there have been several new powerful keymap features and behaviors added, including one of the most asked for features, combos!
+
+#### Combos
+
+The initial [combos](/docs/behaviors/combos) work has landed! The amazing [okke-formsma] has once again delivered another powerful feature for ZMK. Combos are "position based", and are configured in a new toplevel node next to they `keymap` node in user's keymap files.
+
+An example, that would send the `ESC` keycode when pressing both the first and second positions on your keyboard:
+
+```
+/ {
+    combos {
+        compatible = "zmk,combos";
+        combo_esc {
+            timeout-ms = <50>;
+            key-positions = <0 1>;
+            bindings = <&kp ESC>;
+        };
+    };
+};
+```
+
+:::note
+Combos currently are "global", and not scoped to a given active layer. There is future planned work to allow enabling a certain combo for only certain active layers.
+:::
+
+#### Sticky Keys (One-Shot Mods/Layers)
+
+[okke-formsma] also contributed the initial "sticky keys" behavior, which can be used for functionality sometimes called "one shot mods" or "one shot layers". In your keymap, this would like like:
+
+```
+&sk LEFT_CONTROL
+```
+
+for a sticky key/modifier, or:
+
+```
+&sl NAV
+```
+
+for a sticky layer.
+
+#### `&to` Layer Behavior
+
+[mcrosson] contributed the new [`&to`](/docs/behaviors/layers#to-layer) layer related behavior. This can be used to completely replace the active layer with a new one.
+
+This is most frequently used when using multiple core base layers with different layouts, e.g. QWERTY and DVORAK, to switch between them.
+
+#### Grace-Escape
+
+[okke-formsma] added an implementation of the "Grave Escape" behavior, developing a more generic "mod-morph" behavior to do so. Adding
```

Comment:
Should we link to documentation on mod-morph here? (Is there documentation for mod-morph/grave escape? I didn't find anything when I searched.)

Reply:
Agreed, I actually implemented a tweaked version of the grave-esc for my personal keymaps that does more of what i wanted from grave-esc than the defaults.

Reply:
No docs for this yet, sadly, which is why I added a brief intro on how to use it here.